service/frontend/wrappers/metered/api_generated.go

```diff
@@ -319,9 +319,9 @@ func (h *apiHandler) PollForDecisionTask(ctx context.Context, pp1 *types.PollFor
 	defer func() { log.CapturePanic(recover(), h.logger, &err) }()
 	tags := []tag.Tag{tag.WorkflowHandlerName("PollForDecisionTask")}
 	tags = append(tags, toPollForDecisionTaskRequestTags(pp1)...)
-	scope := h.metricsClient.Scope(metrics.FrontendPollForDecisionTaskScope).Tagged(metrics.DomainTag(pp1.GetDomain())).Tagged(metrics.GetContextTags(ctx)...)
-	scope.IncCounter(metrics.CadenceRequests)
-	sw := scope.StartTimer(metrics.CadenceLatency)
+	scope := common.NewPerTaskListScope(pp1.Domain, pp1.TaskList.GetName(), pp1.TaskList.GetKind(), h.metricsClient, metrics.FrontendPollForDecisionTaskScope).Tagged(metrics.GetContextTags(ctx)...)
+	scope.IncCounter(metrics.CadenceRequestsPerTaskList)
+	sw := scope.StartTimer(metrics.CadenceLatencyPerTaskList)
```

Comment:
this might deserve a rollup value, as we'll lose domain-wide aggregates with this change

Reply:
Agree. I also though that might be a problem and based on [this line](https://github.com/uber/cadence/blob/a711217807fb05102232b1dd08aa0f492cd6d114/common/metrics/defs.go#L2804) it looks like there's rollup defined for this metric.

Reply:
UPDATED! Also fixed a bug in metrics package.

Reply:
ah, just doing a manual non-TL-containing timer.  yea, that seems fine 👍